lms/djangoapps/learner_dashboard/views.py

```diff
@@ -47,3 +51,53 @@ def program_details(request, program_uuid):
     }
 
     return render_to_response('learner_dashboard/program_details.html', context)
+
+
+@view_auth_classes()
+class ProgramDiscussionIframeView(APIView):
+    """
+    A view for retrieving Program Discussion IFrame .
+
+    Path: ``/dashboard/dashboard/programs/{program_uuid}/iframe/``
```

Comment:
the path can be improved here and be more meaningful. 
e.g. `/dashboard/programs/{program_uuid}/discussion/`